src/main/java/Parser.java

```diff
@@ -0,0 +1,78 @@
+public class Parser {
+    public static Task parseTask(String type, String input) {
+        int descriptionIndex;
+        String description;
+        Task newTask;
+
+        switch (type) {
+        case InputManager.TODO_PHRASE:
+            descriptionIndex = input.indexOf(InputManager.TODO_PHRASE);
+
+            // add one to remove space
+            description = input.substring(descriptionIndex + InputManager.TODO_PHRASE.length() + 1,
+                    input.length());
+
+            newTask = new Todo(description);
+            break;
+        case InputManager.DEADLINE_PHRASE:
+            descriptionIndex = input.indexOf(InputManager.DEADLINE_PHRASE);
+            int byIndex = input.indexOf(InputManager.BY_PHRASE);
+
+            // add one to remove space
+            // minus one to remove space
+            description = input.substring(
+                    descriptionIndex + InputManager.DEADLINE_PHRASE.length() + 1, byIndex - 1);
+            String by =
+                    input.substring(byIndex + InputManager.BY_PHRASE.length() + 1, input.length());
+
+            newTask = new Deadline(description, by);
+            break;
+        case InputManager.EVENT_PHRASE:
```

Comment:
Personally, I prefer a short code for switch statements. You can create different function to handle each case. But I think it should be alright